gsl/gsl_util

```diff
@@ -135,30 +135,30 @@ inline T narrow(U u)
 // at() - Bounds-checked way of accessing static arrays, std::array, std::vector
 //
 template <class T, size_t N>
-constexpr T& at(T (&arr)[N], size_t index)
+constexpr T& at(T (&arr)[N], std::ptrdiff_t index)
 {
-    Expects(index < N);
-    return arr[index];
+    Expects(index >= 0 && index < narrow_cast<decltype(index)>(N));
+    return arr[static_cast<size_t>(index)];
```

Comment:
> index is already defined so if it gets changed again in the future, you don't have to worry about the cast (i.e. you only have to change one). 

I suspected that but wasn't sure.
I actually prefer the simpler `std::ptrdff_t` as it makes obvious whether the cast is meaningful. 
I don't expect anyone to change back to `std::size_t`, but if someone does for some reason, the `decltype` version won't call attention -- because "you have to change only one place" and "it is obviously correct".  This is one of the very few places where I see "redundancy" as feature.